[Blog] Add the Pulsar 2022 Year in Review blog

Sherlock113 opened the pull request:
Add the Pulsar 2022 Year in Review blog.

Previews:
![Screen Shot 2023-01-10 at 18 26 45](https://user-images.githubusercontent.com/65327072/211526681-a05c5228-eb41-4bcf-af9d-8a73833e7486.png)
![Screen Shot 2023-01-10 at 18 26 59](https://user-images.githubusercontent.com/65327072/211526693-5221638c-77f8-4ccc-87ac-2c1a6f4016e0.png)
![Screen Shot 2023-01-10 at 18 27 06](https://user-images.githubusercontent.com/65327072/211526713-593becf7-9a9e-47e7-a0a1-f45e607c7648.png)
![Screen Shot 2023-01-10 at 18 27 12](https://user-images.githubusercontent.com/65327072/211526733-84ab6ef8-6f62-46f4-9d0f-757e854335bf.png)
![Screen Shot 2023-01-10 at 18 27 18](https://user-images.githubusercontent.com/65327072/211526747-695f65e7-eea9-4c91-9e38-c1875199ae36.png)

Sherlock113 commented:
@merlimat @eolivelli @dave2wave @tisonkun @codelipenghui PTAL. If you think there is sth not proper for this blog, especially the ecosystem parts, please tell me and I will adjust/remove them. Thanks.

tuhaihe reviewed (approved):
LGTM. Thanks @Sherlock113 !

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+
+In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/102 ...
+
+- [2.7.5](https://github.com/apache/pulsar/releases/tag/v2.7.5)
```
Use reverse order here? Then users can read the latest versions first.

Sherlock113 replied:
Changed.

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+
+Updates about clients, Pulsar Manager, and Pulsar Helm Chart are listed below:
+
+- [Pulsar C++ Client 3.0.0](https://github.com/apache/pulsar-client-cpp/releases/tag/v3.0.0)
```
ditto

Sherlock113 replied:
Thanks. Changed.

tisonkun reviewed (commented):
Generally looks good. Thank you! Comments inline.

eolivelli reviewed (changes requested):
Overall it looks good,
but as you say, probably the Ecosystem part is focused mostly on releases of a single vendor.

If we want to cite products delivered by third parties we should define a clear policy (for instance at DataStax, my employer we also released a lot of things for the Pulsar Ecosystem, but there are probably other examples of third party OSS products from other vendors)

I suggest to keep only references to well known and independent communities, like Apache Flink and Spring.io

tisonkun commented:
> keep only references to well-known and independent communities

Perhaps plus a link to the [ecosystem page](https://pulsar.apache.org/ecosystem/) :)

lhotari commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+- [Pulsar Python Client 3.0.0](https://github.com/apache/pulsar-client-python/releases/tag/v3.0.0)
+- [Pulsar Manager 0.3.0](https://github.com/apache/pulsar-manager/releases/tag/v0.3.0)
+- [Pulsar Helm Chart 3.0.0](https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-3.0.0)
+
```
Please add a reference to the "Reactive client for Apache Pulsar 0.1.0" release:
```suggestion
 - [Reactive client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)
 
```

Sherlock113 replied:
Added. Thanks.

lhotari commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+
+In 2022, the Pulsar community worked with other open-source communities to add more integrations to the Pulsar ecosystem. Notable integrations include:
+
+- [Spring for Apache Pulsar](https://spring.io/blog/2022/09/20/spring-for-apache-pulsar-0-1-0-m1-is-now-available): This project aims to provide Spring-friendly APIs, building blocks, and programming models for writing Java applications that interact with Apache Pulsar. Read this [blog](https://streamnative.io/blog/release/2022-09-21-announcing-spring-for-apache-pulsar/) to see details.
```
The StreamNative blog post is outdated. I'd suggest providing a link to the official Spring blog post about 0.1.0 release. /cc @cbornet 
```suggestion
- [Spring for Apache Pulsar](https://spring.io/blog/2022/09/20/spring-for-apache-pulsar-0-1-0-m1-is-now-available): This project aims to provide Spring-friendly APIs, building blocks, and programming models for writing Java applications that interact with Apache Pulsar. Please check the blog post [Spring for Apache Pulsar 0.1.0 available now](https://spring.io/blog/2022/12/15/spring-for-apache-pulsar-0-1-0-available-now) for most latest details and references to documentation.
```

Sherlock113 replied:
Updated. Thanks.

cbornet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+
+## Pulsar ecosystem
+
+In 2022, the Pulsar community worked with other open-source communities to add more integrations to the Pulsar ecosystem. Notable integrations include:
```
I suggest to add the HTTP Sink connector introduced by [PIP 208](https://github.com/apache/pulsar/issues/17719)

Sherlock113 replied:
Added. Thanks.

michaeljmarshall commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,109 @@
+
+## What’s next in 2023
+
+The Pulsar community seeks to further improve the project from different aspects. For example, we plan on releasing the first long-term support (LTS) version of Pulsar to meet the needs of different users. For those who want a more stable Pulsar version, they can choose the LTS version, which is expected to be released every 18 months with bug fix support for 24 months and security patches for 36 ...
```
I wonder if it is worth changing the language here so that we do not over promise release frequency. The 2.11 release took 4 months, and it is only just wrapping up now. I know that we've discussed targeting frequent releases, and I know that users like predictability, but I think it is better to under promise release frequency here.

Sherlock113 replied:
Thanks. I removed the frequency info here.

Sherlock113 commented:
I removed the ecosystem tools provided by vendor and updated the blog. Thank you all for your review! Let me know if you have other questions. @tisonkun @eolivelli @lhotari @cbornet @michaeljmarshall

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+
+## Project releases
+
+In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/102 ...
```
```suggestion
In 2022, the Pulsar community worked hard to improve the project’s capabilities and fix existing bugs with 1 major version and 8 minor versions. Many important Pulsar capabilities were delivered in these releases, such as [cluster-level automatic failure recovery](https://github.com/apache/pulsar/pull/13316) and [lazy-loading added to PartitionedProducer](https://github.com/apache/pulsar/pull/10279). Check out each release below to see details:
```

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [2.8.2](https://github.com/apache/pulsar/releases/tag/v2.8.2)
+- [2.7.5](https://github.com/apache/pulsar/releases/tag/v2.7.5)
+
+For more information, see the [Release Notes page](https://pulsar.apache.org/release-notes/).
```
```suggestion
For more information, see the [Release Notes page](pathname:///release-notes/).
```

... render an internal link for local debugging.

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [Pulsar Manager 0.3.0](https://github.com/apache/pulsar-manager/releases/tag/v0.3.0)
+- [Pulsar Helm Chart 3.0.0](https://github.com/apache/pulsar-helm-chart/releases/tag/pulsar-3.0.0)
+
+For more information, see the [Clients Release Notes page](https://pulsar.apache.org/release-notes/clients/).
```
```suggestion
For more information, see the [Clients Release Notes page](pathname:///release-notes/clients/).
```

ditto

tisonkun commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [Flink-Pulsar Sink Connector](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/pulsar/#pulsar-sink): The Flink-Pulsar Sink Connector is part of the Flink-Pulsar DataStream Connector. It implements Flink’s new [SinkV2 API](https://cwiki.apache.org/confluence/display/FLINK/FLIP-177%3A+Extend+Sink+API) and allows you to write Flink job results back to Pulsar topics s ...
+- [HTTP Sink Connector](https://github.com/apache/pulsar/issues/17719): The HTTP Sink Connector pushes records to any HTTP server with the record value in the body of POST method. The body of the HTTP request is the JSON representation of the record value.
+
+For more information, see the [Ecosystem page](https://pulsar.apache.org/ecosystem/).
```
```suggestion
For more information, see the [Ecosystem page](pathname:///ecosystem/).
```

ditto

tisonkun reviewed (approved):
LGTM. Minor comments inline.

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- **598** global contributors for the Pulsar main GitHub repository
+- **12.1K+** GitHub stars
+- **3.2K+** forks
+- **16** new Committers
+- **5** new PMC members
```
1. The NO. here is a bit confusing. Some are `total NO.` while others are `NO. of last year`. Does it make sense to add the `total NO. of committers and PMC members`? 

In this way, we can reduce users' confusion (data is consistent) and show the whole picture to them.

2. Does it make sense to add the `NO. of Pulsar releases` (show activeness and long history) and the `NO. of ecosystem plugins` (show diversity and compatibility)? 

These data can give users more trust and confidence in Pulsar.

Sherlock113 replied:
1. This is a summary blog for 2022. The only reason I added a total number is that I am not sure the exact number of growth from last year so I added the total number. @tuhaihe If you have the number please let me know. Thanks.
2. For releases, people can find the info in the sections below. For ecosystem tools, you can have a look at the discussions in this pr. It is difficult to include all tools as we may have different vendors developing them.

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+---
+author: Apache Pulsar Community
```
```suggestion
author: Sherlock Xu
```
You deserve it 😊

Sherlock113 replied:
Thank you, but I refuse to do this. This is a summary blog for Pulsar achievements over the past year made possible by the community.

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [New York City Apache Pulsar Meetup](https://www.meetup.com/new-york-city-apache-pulsar-meetup/)
+- [SF Bay Area Apache Pulsar Meetup](https://www.meetup.com/sf-bay-area-apache-pulsar-meetup/)
+- [Netherlands Apache Pulsar Meetup](https://www.meetup.com/netherlands-apache-pulsar-meetup/)
+- [Japan Pulsar User Group](https://japan-pulsar-user-group.connpass.com/)
```
May we know why not add the following groups?

• [SoCal Apache Pulsar Neighborhood Meetup Group](https://www.meetup.com/socal-apache-pulsar-meetup-group/)
• [Beijing Apache Pulsar Meetup](https://www.meetup.com/beijing-apache-pulsar-meetup-by-streamnative/)
• [NorCal Apache Pulsar Neighborhood Meetup Group](https://www.meetup.com/nor-cal-apache-pulsar-meetup-group/)

Sherlock113 replied:
@tuhaihe

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [Pulsar Go Client 0.9.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.9.0)
+- [Pulsar Go Client 0.8.1](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.1)
+- [Pulsar Go Client 0.8.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.0)
+- [Pulsar Node.js client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
```
```suggestion
- [Pulsar Node.js Client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
```

Keep consistent

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [Pulsar Go Client 0.8.1](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.1)
+- [Pulsar Go Client 0.8.0](https://github.com/apache/pulsar-client-go/releases/tag/v0.8.0)
+- [Pulsar Node.js client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
+- [Pulsar Node.js client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)
```
```suggestion
- [Pulsar Node.js Client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)
```

Anonymitaet commented on blog/2023-01-10-pulsar-2022-year-in-review.md:
```diff
@@ -0,0 +1,106 @@
+- [Pulsar Node.js client 1.7.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.7.0)
+- [Pulsar Node.js client 1.6.2](https://github.com/apache/pulsar-client-node/releases/tag/v1.6.2)
+- [Pulsar Python Client 3.0.0](https://github.com/apache/pulsar-client-python/releases/tag/v3.0.0)
+- [Reactive client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)
```
```suggestion
- [Reactive Client for Apache Pulsar 0.1.0](https://github.com/apache/pulsar-client-reactive/releases/tag/v0.1.0)
```

eolivelli reviewed (approved):
Thanks for addressing my comments.
LGTM at the current status

tisonkun commented:
Merging...

Thanks for your contribution @Sherlock113 and thanks to all reviewers!
